src/macros.rs

```diff
@@ -1,30 +1,72 @@
 #[macro_export]
-macro_rules! commands_enum {
-    // Case when command has aliases (e.g. add(a, b, c))
-    ($($module:ident $(($($alias:ident),*))?),*) => (
+macro_rules! commands {
+    ($($module:ident $(($($alias:ident),*))?),*) => {
         paste::paste! {
-            #[derive(Subcommand)]
-            enum Commands {
+            /// Build the global CLI (root command) and attach module subcommands.
+            pub fn build_args() -> clap::Command {
+                // Use your desired root name here (for example, "railway" rather than a derived name).
+                let mut cmd = clap::Command::new("railway")
+                    .about("Railway CLI")
+                    .author(clap::crate_authors!())
+                    .propagate_version(true)
+                    .about(clap::crate_description!())
+                    .long_about(None)
+                    .version(clap::crate_version!())
+                    .arg(
+                        clap::Arg::new("json")
+                            .long("json")
+                            .action(clap::ArgAction::SetTrue)
+                            .help("Output in JSON format")
+                    );
                 $(
-                    #[clap(
-                        $(visible_aliases = &[$( stringify!($alias) ),*])?
-                    )]
-                    [<$module:camel>]($module::Args),
+                    {
+                        // Get the subcommand as defined by the module.
+                        let sub = <$module::Args as ::clap::CommandFactory>::command();
+                        // Allow the module to add dynamic arguments (if needed) and add any aliases.
+                        let sub = {
+                            let mut s = sub;
+                            $(
+                               s = s.visible_alias($( stringify!($alias) ),*);
+                            )?
+                            #[allow(unused_imports)]
+                            {
+                                // First import any definitions from the module.
+                                // Then import everything (including the fallback).
+                                use $crate::commands::get_dynamic_args;
+                                {
+                                    use $crate::commands::$module::*;
+                                    s = get_dynamic_args(s);
+                                }
+                            }
+                            s = s.name(stringify!($module));
+                            s
+                        };
+                        // Add this subcommand into the global CLI.
+                        cmd = cmd.subcommand(sub);
```

Comment:
Is this macro necessary? In general, I would avoid making macros this big. Its hard to follow and not very maintainable.

Reply:
This macro was always how commands are defined